### DISCUSSION TITLE
UPS Output Voltage Reading Incorrect?

### DISCUSSION THREAD

**Discussion by DaGwhite** [nutdotnet/WinNUT-Client#139] [Q&A]
I just got the program up and running.  I noticed the default calibration settings were geared for European power numbers, and I changed the settings for US power; 60 Hz, and the voltage ranges going from 100 to 140. 

Everything looks fine now, except the UPS output voltage is displayed as 136Vac instead of 120.  I checked the front panel on the UPS (a CyberPower CP1500PFCLCDa), and it is reading 121Vac for output voltage.  The UPS is connected via USB to a NetGear ReadyNAS 312 NAS, which is reporting the UPS info to WinNUT.  The control software for the NAS doesn't report the UPS output voltage, so it's hard to tell where in the chain of multiple bits of hardware-to-software-to-software-to-more-software the voltage info is getting corrupted.

I did a search, and didn't find reports of similar issues with WinNUT.  Any ideas on how to chase this down?

Thanks!

**Comment by gbakeman** [comment]
Hey @DaGwhite ,

I think the issue you're looking for is #101. It sounds like there's a bug with the driver NUT uses to communicate with the UPS. We're debating creating some kind of special override, although it sounds like the NUT team may have fixed it in NUT server version 2.8.0. See if you can somehow update the NUT server on your ReadyNAS, or you may need another instance of NUT that's more up-to-date.

**Comment by DaGwhite** [comment]
Thanks!  I am running the latest firmware, but this is an older NAS, and it's not clear that they will bother updating it.  Installing a new NUT server is over my head, and I may just have to live with it.  I just checked, and it also looks like NetGear is out of the NAS business.

**Reply by gbakeman** [reply]
I was afraid it may be abandoned. I can't rule out having more fine-grained controls in a future WinNUT version, stay tuned!

**Comment by DaGwhite** [comment]
From checking out the other postings, apparently Synology NAS's can have the same issue.  Someday I will probably need a new NAS, and hopefully this will all be resolved by then.  

In the meantime, I can live with it.  Thanks for the slick program!

**Comment by Eskeu** [comment]
I have a Synology DS220+ (released in 2020) connected to an APC SRT1500RMXLA.  Earlier versions incorrectly reported it connected at 240V / 50hz when it was actually plugged in to 120v /60hz (I'm in the United States).  I just upgraded to Pre-Release v2.3.8834 hoping it would correct this.  Now the load, input, and output are all showing zero values.  Battery voltage and input frequency are correct though.

**Reply by gbakeman** [reply]
It looks like the issue you're experiencing is related to #116 . Previously, WinNUT would substitute in the nominal value when there was an error retrieving a variable (such as it didn't exist on the NUT server.) Recently, I modified WinNUT to report an out of range value (-1/0) when a variable is unavailable. There's still work to do to handle this better, but at least the user is more informed when something is wrong. You can look in the Variable List window (under the File menu) to see what the NUT server is reporting to WinNUT. If you're expecting more variables than what you're seeing, then I'm afraid that you may need to troubleshoot the NUT server.
